### DISCUSSION TITLE
Templating Git Sync directory

### DISCUSSION THREAD

**Discussion by RaghavBatra** [apache/airflow#24510] [General]
### Official Helm Chart version

1.6.0 (latest released)

### Apache Airflow version

2.3.0

### Kubernetes Version

1.22.6

### What happened

We use the official Airflow Helm Chart for deploying our production Airflow instances. 
However, whenever going from local to our dev instance, we run into a path issue since the GIT_SYNC_DEST field is hard-coded in the `_helpers.yaml` file (line 211). Because of the hard-coded value of "repo", all our local instance paths have to be changed to account for this. 

For example, `dags/example_dag.py` becomes `dags/repo/example_dag.py`. This becomes very tedious very quickly.
Reproducing the field below:
```
    - name: GIT_SYNC_DEST
      value: "repo"
```

### What you think should happen instead

 Is it possible to template this value so we can set to whatever works for each team?
Something like:

```
    - name: GIT_SYNC_DEST
      value: {{ .Values.dags.gitSync.dest | quote }}
```

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
Assigned you if you want to make PR.

**Comment by jedcunningham** [comment]
Shouldn't you be using `[core] dags_folder` to construct the path anyways? That'll be more portable.

**Comment by RaghavBatra** [comment]
@jedcunningham the issue is that because `repo` is hard-coded, it will always be an additional folder, regardless of `dags_folder`.

**Reply by jedcunningham** [reply]
Check the value of `[core] dags_folder` when you have gitsync on... That points at the `repo` level, not its parent. That's what I mean by portable - you should construct paths with dags_folder if you need the absolute path, as that can vary Airflow instance by Airflow instance.

**Comment by potiuk** [comment]
Thisis at most a discussion. I am quite lost what you want to do. 

Can you please explain what is the structure of you repos/teams/directories @RaghavBatra. Can you also please explai what did you try to get it working for you ? differnet repos? Submodules? difererent subdirs.

I have a feeling (though I mighjt be wrong because you have not explained enough how you tried and failed to used the current mode) that you are trying to use git-sync in the way that it has not been intended.

Looking forward to more explanation. Please also bear in mind that we do not know your particular case and you should explain in detail your case, and ideally do it in the way that we can understand it without asking too many questions. You can watch the talk of mine https://www.youtube.com/watch?v=G6VjYvKr2wQ where I am asking both users and maintainer for empathy and one of the asks there is that users of free software should make an effort to understand that the ratio of maintainers to users is 1000: 1 and that they should simply make reasonable and substantial effort to explain their case when they want some help in their case.

Looking for a detailed effort from your side @RaghavBatra  to explain the case of yours.
